camera.go

```diff
@@ -265,38 +255,66 @@ func (c *Camera) raycast() {
 	combSort(c.spriteOrder, c.spriteDistance, numSprites)
 
 	//after sorting the sprites, do the projection and draw them
-	for i := 0; i < numSprites; i++ {
-		wg.Add(1)
-		go c.asyncCastSprite(i, &wg)
-	}
+	c.asyncCastSprites(numSprites, &wg)
 
 	wg.Wait()
 }
 
 func (c *Camera) asyncCastLevel(levelNum int, wg *sync.WaitGroup) {
-	defer wg.Done()
-
 	rMap := c.mapObj.Level(levelNum)
+	stride := c.w / 100
+	if stride < 1 {
+		stride = 1
+	}
 
-	for x := 0; x < c.w; x++ {
-		c.castLevel(x, rMap, c.levels[levelNum], levelNum, wg)
+	for i := 0; i < c.w; i += stride {
+		wg.Add(1)
+
+		go func(start int) {
+			end := start + stride
+			if end > c.w {
+				end = c.w
+			}
+
+			for x := start; x < end; x++ {
+				c.castLevel(x, rMap, c.levels[levelNum], levelNum)
+			}
+
+			wg.Done()
+		}(i)
 	}
 }
 
-func (c *Camera) asyncCastSprite(spriteNum int, wg *sync.WaitGroup) {
+func (c *Camera) asyncCastSprites(numSprites int, wg *sync.WaitGroup) {
 	defer wg.Done()
+	wg.Add(1)
+
+	stride := numSprites / 32
+	if stride < 1 {
+		stride = 1
+	}
 
-	c.semaphore <- struct{}{} // Lock
-	defer func() {
-		<-c.semaphore // Unlock
-	}()
+	for i := 0; i < numSprites; i++ {
```

Comment:
should be `for i := 0; i < numSprites; i += stride {`

I'll go ahead and fix this in the ovk_performance branch prior to getting it ready to merge to main.

Reply:
Yes, it should be, good catch.